libpromises/rlist.c

```diff
@@ -325,18 +325,50 @@ Rlist *RlistAppendRval(Rlist **start, Rval rval)
     return rp;
 }
 
-Rval RvalNew(const void *item, RvalType type)
+Rval RvalNewRewriter(const void *item, RvalType type, JsonElement *map)
 {
+    int max_size = 10*CF_BUFSIZE+1;
+    char buffer_from[max_size];
+    char buffer_to[max_size];
```

Comment:
This is almost 100KB of stack space. Considering the possibility of CFEngine being used on embedded platforms, I would make this a heap allocation instead, since stack space can be considerably more limited on some platforms.